middleware/xatmi/source/xatmi/extended.cpp

```diff
@@ -144,6 +144,12 @@ const uuid_t* casual_execution_id_get()
    return &casual::common::execution::id().underlying().get();
 }
 
+const uuid_t* casual_execution_id_reset()
+{
+   casual::common::execution::reset();
```

Comment:
I have a vague memory that this function mucks around with environment variables in a way that is susceptible to memory leaks. Looks like it's mentioned in 9dd06ca. Not sure how applicable that issue is here, thoughts @lazan?

Reply:
Yeah, exactly. This will leak memory on linux. But, I think the usage of _set environment_ in `execution::reset()` is a bug we need to fix regardless of exposing c-api to reset the execution-id. 
I'll create a separate issue regarding that (we might have one)

Reply:
Created #275.